src/apisix/t/bk-header-rewrite.t

```diff
@@ -0,0 +1,311 @@
+                    },
+                    "uri": "/hello"
+                }]]
+                )
+
+            if code >= 300 then
+                ngx.status = code
+            end
+            ngx.say(body)
+        }
+    }
+--- request
+GET /t
+--- response_body
+passed
+
+=== TEST 5: set route(rewrite headers)
+--- config
+    location /t {
+        content_by_lua_block {
+            local t = require("lib.test_admin").test
+            local code, body = t('/apisix/admin/routes/1',
+                 ngx.HTTP_PUT,
+                 [[{
+                        "plugins": {
+                            "bk-proxy-rewrite": {
+                                "uri": "/uri/plugin_proxy_rewrite"
+                            },
+                            "bk-header-rewrite": {
+                                "set": {"X-Api-Version": "v2"}
+                            }
+                        },
+                        "upstream": {
+                            "nodes": {
+                                "127.0.0.1:1980": 1
+                            },
+                            "type": "roundrobin"
+                        },
+                        "uri": "/hello"
+                }]]
+                )
+
+            if code >= 300 then
+                ngx.status = code
+            end
+            ngx.say(body)
+        }
+    }
+--- request
+GET /t
+--- response_body
+passed
+
+
+=== TEST 6: rewrite headers
+--- request
+GET /hello HTTP/1.1
+--- more_headers
+X-Api-Version:v1
+--- response_body
+uri: /uri/plugin_proxy_rewrite
+host: localhost
+x-api-version: v2
+x-real-ip: 127.0.0.1
+
+=== TEST 7: set route(rewrite empty headers)
+--- config
+    location /t {
+        content_by_lua_block {
+            local t = require("lib.test_admin").test
+            local code, body = t('/apisix/admin/routes/1',
+                 ngx.HTTP_PUT,
+                 [[{
+                        "plugins": {
+                            "bk-proxy-rewrite": {
+                                "uri": "/uri/plugin_proxy_rewrite"
+                            },
+                            "bk-header-rewrite": {
+                                "set": {"X-Api-Test": "hello"}
+                            }
+                        },
+                        "upstream": {
+                            "nodes": {
+                                "127.0.0.1:1980": 1
+                            },
+                            "type": "roundrobin"
+                        },
+                        "uri": "/hello"
+                }]]
+                )
+
+            if code >= 300 then
+                ngx.status = code
+            end
+            ngx.say(body)
+        }
+    }
+--- request
+GET /t
+--- response_body
+passed
+
+=== TEST 8: rewrite empty headers
+--- request
+GET /hello HTTP/1.1
+--- more_headers
+X-Api-Test:
+--- response_body
+uri: /uri/plugin_proxy_rewrite
+host: localhost
+x-api-test: hello
+x-real-ip: 127.0.0.1
+
+
```

Comment:
- need one case for `add`
- request(foo:foo) + add({hello:world, foo:bar}) = {hello:world, foo:foo bar}